internal/controller/common/numaflowtypes/isbsvc.go

```diff
@@ -0,0 +1,46 @@
+package numaflowtypes
+
+import (
+	"context"
+	"fmt"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaplane/internal/util/kubernetes"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
```

Comment:
created a new directory called `numaflowtypes` which has logic for each of the types which could theoretically be used in multiple places (`numaflowtypes/pipeline` functions are in fact)